packages/renderer/src/components/Modals/PublishProject/component.tsx

```diff
@@ -1,183 +1,51 @@
-          ? t('modal.publish_project.title', { title: project.title })
-          : ''
-      }
-      onClose={handleClose}
-      size="small"
-      onBack={step !== 'initial' ? handleChangeStep('initial') : undefined}
-    >
-      {step === 'initial' && <Initial onStepChange={handleChangeStep} />}
-      {step === 'alternative-servers' && <AlternativeServers onClick={handleClickPublish} />}
-      {step === 'publish-to-land' && <PublishToLand onClose={onClose} />}
-      {step === 'publish-to-world' && <PublishToWorld onClose={onClose} />}
-    </Modal>
-  );
-}
-
-function Initial({ onStepChange }: { onStepChange: (step: Step) => () => void }) {
-  return (
-    <div className="Initial">
-      <span className="select">{t('modal.publish_project.select')}</span>
-      <div className="options">
-        <OptionBox
-          thumbnailSrc={WorldsPng}
-          title={t('modal.publish_project.worlds.title')}
-          description={t('modal.publish_project.worlds.description')}
-          buttonText={t('modal.publish_project.worlds.action')}
-          onClickPublish={onStepChange('publish-to-world')}
-          learnMoreUrl="https://docs.decentraland.org/creator/worlds/about/#publish-a-world"
-        />
-        <OptionBox
-          thumbnailSrc={LandPng}
-          title={t('modal.publish_project.land.title')}
-          description={t('modal.publish_project.land.description')}
-          buttonText={t('modal.publish_project.land.action')}
-          onClickPublish={onStepChange('publish-to-land')}
-          learnMoreUrl="https://docs.decentraland.org/creator/development-guide/sdk7/publishing-permissions/#land-permission-options"
-        />
-      </div>
-      <span
-        className="alternative_servers"
-        onClick={onStepChange('alternative-servers')}
-      >
-        {t('modal.publish_project.alternative_servers.title')}
-      </span>
-    </div>
-  );
-}
-
-function AlternativeServers({ onClick }: StepProps) {
-  const [option, setOption] = useState<AlternativeTarget>('test');
-  const [customUrl, setCustomUrl] = useState('');
-  const [error, setError] = useState('');
-
-  const handleClick = useCallback(() => {
-    if (option === 'custom' && !isUrl(customUrl)) {
-      return setError(t('modal.publish_project.alternative_servers.errors.url'));
-    }
-    const value: StepValue = { target: option, value: customUrl };
-    onClick(value);
-  }, [option, customUrl]);
-
-  const handleChangeSelect = useCallback((e: SelectChangeEvent<AlternativeTarget>) => {
-    setOption(e.target.value as AlternativeTarget);
-  }, []);
-
-  const handleChangeCustom = useCallback(
-    (e: ChangeEvent<HTMLInputElement>) => {
-      if (error) setError('');
-      setCustomUrl(e.target.value);
+  }, [setStep, setHistory, onClose]);
+
+  const handleBack = useCallback(() => {
+    const prev = history.pop();
+    setStep(prev || 'initial');
+    setHistory(history => (history.length > 0 ? history.slice(0, -1) : []));
```

Comment:
we could probably avoid `[step, setStep]` by only using `setHistory`

we push the steps to the history array => we use the last entry on the array as the step to render. If there are no entries in the array, we always render `initial`. This way we avoid the double `setStep, setHistory`